specification/dartLangSpec.tex

```diff
@@ -15773,6 +15768,8 @@ \subsection{Yield-Each}
 Let $T_f$ be the element type of $f$
 (\ref{functions}),
 and let $T$ be the static type of $e$.
+It is a compile-time error if $T$ may not be assigned to
+the declared return type of $f$.
 If $f$ is a generator and
```

Comment:
If it's not, `yield*` is a compile-time error, so maybe this "If" is not necessary.

Reply:
The text here is intended to keep working when we get null safety: For instance, it will then prevent `yield* <num>[];` in a synchronous generator returning `Iterable<int>`. Without null safety, the assignability rule allows `yield* e;` where the static type of `e` implements `Iterable<U>` for any `U`, which is consistent with the current implementations.

So I believe this rule will work as intended now and later. Do you see some other location where this property is stated or can be derived? Otherwise this is simply the place where it's specified that it is a compile-time error.

Reply:
It's only the `$f$ is a generator and` part that I'm questioning. If it wasn't a generator function, the `yield*` would already be a compile-time error according to line 15765

Reply:
Right, removed.